alf/networks/actor_networks.py

```diff
@@ -137,28 +135,30 @@ def forward(self, observation, state=()):
         """
 
         observation, state = super().forward(observation, state)
-        self._encoding_net.eval = self.eval
+        self._encoding_net.train(self.training)
         encoded_obs, _ = self._encoding_net(observation)
 
         actions = []
         i = 0
         for layer, spec in zip(self._action_layers, self._flat_action_spec):
-            _info = {}
+            _info = {} if alf.summary.should_summarize_output() else None
             action = layer(encoded_obs, info=_info)
-
-            name = ('summarize_output/' + self.name + '.action_layer.' + str(i)
-                    + '.pre_activation.output_norm')
-            if self.eval:
-                name += ".eval"
-            alf.summary.scalar(name=name, data=_info["pre_activation"].norm())
+            if alf.summary.should_summarize_output():
+                name = ('summarize_output/' + self.name + '.action_layer.' +
+                        str(i) + '.pre_activation.output_norm')
+                if not self.training:
+                    name += ".eval"
+                alf.summary.scalar(
+                    name=name, data=_info["pre_activation"].norm())
```

Comment:
it's better to be average across samples so it can be easily compared if batch size is changed.